### PR TITLE
DROOLS-4569 : XLS editor should have a save button

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-4569

Bundle:
https://github.com/kiegroup/appformer/pull/809
https://github.com/kiegroup/kie-wb-common/pull/2922
https://github.com/kiegroup/drools-wb/pull/1234

### CODE REVIEW

**Comment by Rikkola** [comment]
jenkins please compile downstream build

File: uberfire-services/uberfire-services-api/src/main/java/org/guvnor/common/services/backend/metadata/MetadataServerSideService.java [hunk 1]
```diff
@@ -28,4 +28,8 @@
     Metadata getMetadata(final Path resource);
 
     List<String> getTags(final Path resource);
+
+    Path saveMetadata(final Path resource,
+                      final Metadata metadata,
+                      final String comment);
```
Comment (jomarko) [inline uberfire-services/uberfire-services-api/src/main/java/org/guvnor/common/services/backend/metadata/MetadataServerSideService.java]: Please add a javadoc,  I think for newcomers can be unclear if this method save really just metadata or also `path`.

File: uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/metadata/MetadataServiceImpl.java [hunk 2]
```diff
@@ -67,12 +74,37 @@ public MetadataServiceImpl() {
...
+                                       IOUtils.toByteArray(inputStream),
+                                       setUpAttributes(Paths.convert(resource),
+                                                       metadata),
+                                       commentedOptionFactory.makeCommentedOption(comment));
+            } finally {
+                inputStream.close();
+            }
+        } catch (IOException e) {
+            throw ExceptionUtilities.handleException(e);
+        }
```
Comment (jomarko) [inline uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/metadata/MetadataServiceImpl.java]:
```suggestion
        try (final InputStream inputStream = ioService.newInputStream(resource, StandardOpenOption.READ)) {
            return ioService.write(resource,
                                   IOUtils.toByteArray(inputStream),
                                   setUpAttributes(Paths.convert(resource),
                                                   metadata),
                                   commentedOptionFactory.makeCommentedOption(comment));
        } catch (IOException e) {
            throw ExceptionUtilities.handleException(e);
        }
```

File: uberfire-services/uberfire-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataServiceImplTest.java [hunk 3]
```diff
@@ -75,6 +96,39 @@ public void testGetEmptyTagsNoOtherMetaView() {
...
+        verify(ioService).write(any(org.uberfire.java.nio.file.Path.class),
+                                any(byte[].class),
+                                anyMap(),
+                                eq(comment));
+        verify(is).close();
+    }
+
+    @Test(expected = GenericPortableException.class)
+    public void testSaveMetaDataException() {
+        final InputStream is = new ByteArrayInputStream("hello".getBytes(StandardCharsets.UTF_8));
```
Comment (jomarko) [inline uberfire-services/uberfire-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataServiceImplTest.java]: We need to close also this stream.

**Comment by Rikkola** [comment]
jenkins please restest this

**Comment by Rikkola** [comment]
Test coverage after the merge about 40% :----------)

File: uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/metadata/MetadataServiceImpl.java [hunk 4]
```diff
@@ -67,12 +74,31 @@ public MetadataServiceImpl() {
...
+                               CommentedOptionFactory commentedOptionFactory,
                                SessionInfo sessionInfo) {
         this.ioService = ioService;
         this.configIOService = configIOService;
+        this.commentedOptionFactory = commentedOptionFactory;
         this.sessionInfo = sessionInfo;
     }
 
+    @Override
+    public Path saveMetadata(final Path resource,
```
Comment (jomarko) [inline uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/metadata/MetadataServiceImpl.java]: In interface we use `final Path path` and here `final Path resource`. Could we please unify? To me second option sounds better, however feel free to pick any option.

File: uberfire-services/uberfire-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataServiceImplTest.java (hunk 3 above)
Comment (jomarko) [reply uberfire-services/uberfire-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataServiceImplTest.java]: Can't we really close the stream due to catching an exception, even if we use `org.assertj.core.api.Assertions.assertThatThrownBy()` ?

Comment (Rikkola) [reply uberfire-services/uberfire-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataServiceImplTest.java]: But isn't the stream closed in `saveMetadata`?

Comment (Rikkola) [reply uberfire-services/uberfire-services-backend/src/test/java/org/guvnor/common/services/backend/metadata/MetadataServiceImplTest.java]: I added close, just incase something goes wrong before the close.

**Comment by Rikkola** [comment]
@karreiro I might ask you to check this if that is ok with you.

**Pull request merged**